src/sim/mem_pool.cc

```diff
@@ -148,10 +168,30 @@ MemPools::populate(const AddrRangeList &memories)
         pools.emplace_back(pageShift, mem.start(), mem.end());
 }
 
+Addr
+MemPools::allocatedBytes() const
+{
+    return std::transform_reduce(
+        pools.begin(), pools.end(), static_cast<Addr>(0),
+        std::plus<Addr>(), std::mem_fn(&MemPool::allocatedBytes));
+}
+
 Addr
 MemPools::allocPhysPages(int npages, int pool_id)
 {
-    return pools[pool_id].allocate(npages);
+    const Addr addr = pools[pool_id].allocate(npages);
+
+    // Recompute max allocated bytes stat, since it may have changed.
+    stats.maxAllocatedBytes =
+        std::max<Addr>(stats.maxAllocatedBytes.value(), allocatedBytes());
```

Comment:
I think this needs a bit of attention

When dealing with stats one always needs to take into consideration they can (and will) be reset manually with the m5 reset pseudo op before entering a ROI.

So at a certain point maxAllocatedBytes might be reset to 0.

What is the stat supposed to model? Is it supposed to model the maximum number of allocated bytes in the entire simulation (m5 reset shouldn't affect it)? Or do we want to model the number of allocated bytes in a specific time window (m5 reset should affect it)

Reply:
> m5 reset shouldn't affect it

Let's take this discussion off of this thread, but I'm not sure we should have any "stats" which are not reset. We've had confusion in the past with this. Maybe we should have a different word/interface for stats which don't reset.

See #1615 and #1699 for examples.

Let's discuss this at a high level at the next developers meeting.

Reply:
I removed this stat to simplify the patch, since it's not necessary for the free list to work properly.